File: src/leetcode/python/python/bit_manipulation/MajorityElement_169.py
class MajorityElement_169(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        lens = len(nums)
        major = nums[0]
        count = 1
        for num in nums[1:]:
            if num != major:
                count -= 1
                if count == 0:
                    count = 1
                    major = num
            else:
                count += 1
                if count > lens / 2:
                    break
        return major

File: src/leetcode/java/MajorityElement_169.java
package leetcode.java.easy;

public class MajorityElement_169 {
  public int majorityElement(int[] nums) {
    int cnt = 1;
    int major = nums[0];
    for (int i = 1; i < nums.length; i++) {
      if (nums[i] != major) {
        cnt--;
        if (cnt == 0) {
          major = nums[i];
          cnt = 1;
        }
      } else {
        if (++cnt > nums.length / 2) break;
      }
    }
    return major;
  }
}
